source/fe/fe_data.cc

```diff
@@ -19,9 +19,193 @@
+        default:
+          Assert(false, ExcNotImplemented());
+          return 0;
+      }
+  }
+
+  inline unsigned int
+  hexes_per_cell(const ReferenceCell::Type cell_type)
+  {
+    switch (cell_type)
+      {
+        case ReferenceCell::Type::Vertex:
+          return GeometryInfo<0>::hexes_per_cell;
+        case ReferenceCell::Type::Line:
+          return GeometryInfo<1>::hexes_per_cell;
+        case ReferenceCell::Type::Tri:
+          return 0;
+        case ReferenceCell::Type::Quad:
+          return GeometryInfo<2>::hexes_per_cell;
+        case ReferenceCell::Type::Tet:
+          return 1;
+        case ReferenceCell::Type::Hex:
+          return GeometryInfo<3>::hexes_per_cell;
+        default:
+          Assert(false, ExcNotImplemented());
+          return 0;
+      }
+  }
+
+  inline unsigned int
+  vertices_per_face(const ReferenceCell::Type cell_type)
+  {
+    switch (cell_type)
+      {
+        case ReferenceCell::Type::Vertex:
+          return GeometryInfo<0>::vertices_per_face;
+        case ReferenceCell::Type::Line:
+          return GeometryInfo<1>::vertices_per_face;
+        case ReferenceCell::Type::Tri:
+          return 2;
+        case ReferenceCell::Type::Quad:
+          return GeometryInfo<2>::vertices_per_face;
+        case ReferenceCell::Type::Tet:
+          return 3;
+        case ReferenceCell::Type::Hex:
+          return GeometryInfo<3>::vertices_per_face;
+        default:
+          Assert(false, ExcNotImplemented());
+          return 0;
+      }
+  }
+
+  inline unsigned int
+  lines_per_face(const ReferenceCell::Type cell_type)
+  {
+    switch (cell_type)
+      {
+        case ReferenceCell::Type::Vertex:
+          return GeometryInfo<0>::lines_per_face;
+        case ReferenceCell::Type::Line:
+          return GeometryInfo<1>::lines_per_face;
+        case ReferenceCell::Type::Tri:
+          return 1;
+        case ReferenceCell::Type::Quad:
+          return GeometryInfo<2>::lines_per_face;
+        case ReferenceCell::Type::Tet:
+          return 3;
+        case ReferenceCell::Type::Hex:
+          return GeometryInfo<3>::lines_per_face;
+        default:
+          Assert(false, ExcNotImplemented());
+          return 0;
+      }
+  }
+
+  inline unsigned int
+  quads_per_face(const ReferenceCell::Type cell_type)
+  {
+    switch (cell_type)
+      {
+        case ReferenceCell::Type::Vertex:
+          return GeometryInfo<0>::quads_per_face;
+        case ReferenceCell::Type::Line:
+          return GeometryInfo<1>::quads_per_face;
+        case ReferenceCell::Type::Tri:
+          return 0;
+        case ReferenceCell::Type::Quad:
+          return GeometryInfo<2>::quads_per_face;
+        case ReferenceCell::Type::Tet:
+          return 1;
+        case ReferenceCell::Type::Hex:
+          return GeometryInfo<3>::quads_per_face;
+        default:
+          Assert(false, ExcNotImplemented());
+          return 0;
+      }
+  }
```

Comment:
@drwells The problem are these functions. These access information of the faces, which is not (and I don't want it to be come) part of the interface of `ReferenceCell::Info` (because it is not extendable to mixed meshes). Maybe I could add to the interface a function `get_face_reference_type(unsigned int face)`, which would return the right `ReferenceCell::Type` for which we could query here the number of (sub)entities. What do you think?

This would also for mixed meshes.